src/components/KPop/KPop.vue

```diff
@@ -1,431 +1,306 @@
+  return {
+    width: getSizeFromString(props.width),
+    maxWidth: getSizeFromString(props.maxWidth),
+    maxHeight: getSizeFromString(props.maxHeight),
+    ...floatingStyles.value,
+  }
+})
+
+const popoverClassesObj = computed(() => [props.popoverClasses, { 'hide-caret': props.hideCaret }])
+
+const { floatingStyles, placement: calculatedPlacement } = useFloating(popoverTrigger, popoverElement, {
+  ...(props.placement === 'auto' ? { middleware: [autoPlacement()] } : { placement: props.placement }),
+  strategy: props.positionFixed ? 'fixed' : 'absolute',
+  /**
+   * floating-ui docs don't recommend using whileElementsMounted with v-show
+   * but otherwise the popover won't update its position even when using manual update method
+   * so we keep using autoUpdate for now
+   * Docs: https://floating-ui.com/docs/vue#anchoring
+   */
+  whileElementsMounted: autoUpdate,
+})
+
+onMounted(() => {
+  if (document) {
+    document.addEventListener('click', clickHandler)
+
+    if (popoverTrigger.value) {
+      if (props.trigger === 'hover') {
+        popoverTrigger.value.addEventListener('mouseenter', showPopover)
+        popoverTrigger.value.addEventListener('focus', showPopover)
+        popoverTrigger.value.addEventListener('mouseleave', hidePopover)
+        popoverTrigger.value.addEventListener('blur', hidePopover)
+      } else {
+        popoverTrigger.value.addEventListener('click', togglePopover)
       }
-    },
-    bindEvents() {
-      const popper = this.$refs.popper
-      if (popper) {
-        if (this.trigger === 'hover') {
-          this.reference.addEventListener('mouseenter', this.createInstance)
-          this.reference.addEventListener('focus', this.createInstance)
-          this.reference.addEventListener('mouseleave', this.hidePopper)
-          this.reference.addEventListener('blur', this.hidePopper)
-          popper.addEventListener('mouseenter', this.showPopper)
-          // this is important event listener that allows to keyboard navigate through the popover without closing it
-          popper.addEventListener('focusin', this.showPopper)
-          popper.addEventListener('mouseleave', this.hidePopper)
-          // has to come in pair with focusin
-          popper.addEventListener('focusout', this.hidePopper)
-        }
+    }
+
+    if (popoverElement.value) {
+      popoverElement.value.addEventListener('click', onPopoverClick)
 
-        popper.addEventListener('click', this.onPopperContentClick)
+      if (props.trigger === 'hover') {
+        popoverElement.value.addEventListener('mouseenter', showPopover)
+        popoverElement.value.addEventListener('focusin', showPopover)
+        popoverElement.value.addEventListener('mouseleave', hidePopover)
+        popoverElement.value.addEventListener('focusout', hidePopover)
       }
+    }
+  }
+})
 
-      if (document) {
-        document.documentElement.addEventListener('click', this.handleClick)
+onBeforeUnmount(() => {
+  if (document) {
```

Comment:
onBeforeMount isn't called during SSR, so you shouldn't need the check. Doesn't hurt to leave it I guess